controllers/provider-alicloud/pkg/webhook/controlplanebackup/ensurer.go

```diff
@@ -96,50 +98,54 @@ func (e *ensurer) getBackupRestoreContainer(name string, cluster *extensionscont
 	// They are only specified for the etcd-main stateful set (backup is enabled)
 	var (
 		provider                string
+		prefix                  string
 		env                     []corev1.EnvVar
 		volumeClaimTemplateName = name
 	)
 	if name == gardencorev1alpha1.StatefulSetNameETCDMain {
-		provider = alicloud.StorageProviderName
-		env = []corev1.EnvVar{
-			{
-				Name: "STORAGE_CONTAINER",
-				// The bucket name is written to the backup secret by Gardener as a temporary solution.
-				// TODO In the future, the bucket name should come from a BackupBucket resource (see https://github.com/gardener/gardener/blob/master/docs/proposals/02-backupinfra.md)
-				ValueFrom: &corev1.EnvVarSource{
-					SecretKeyRef: &corev1.SecretKeySelector{
-						Key:                  alicloud.BucketName,
-						LocalObjectReference: corev1.LocalObjectReference{Name: alicloud.BackupSecretName},
+		if cluster.CloudProfile.Spec.Backup == nil {
+			e.logger.Info("Backup profile is not configured;  backup will not be taken for etcd-main")
```

Comment:
Can you return immediately instead of having `else` branch? Would make the code better readable (same for other provider ensurer, please).

Reply:
Umm. You want me repeat "return controlplane.GetBackupRestoreContainer(name, volumeClaimTemplateName, schedule, provider, prefix, image.String(), nil, env, nil), nil"
statement multiple times?

Reply:
No strong opinion about it, just something that came to my mind. If you want, then do it, otherwise let's keep it as it is :)

Reply:
Its not about efforts changing it but I found this way more readable. And less error prone for any future changes in `controlplane.GetBackupRestoreContainer(name, volumeClaimTemplateName, schedule, provider, prefix, image.String(), nil, env, nil), nil`. Single place for change is good i feel. So, i'll keep it as it is then.

Reply:
OK 👍